arbnode/sequencer.go

```diff
@@ -508,6 +598,8 @@ func (s *Sequencer) Start(ctxIn context.Context) error {
 	}
 
 	s.CallIteratively(func(ctx context.Context) time.Duration {
+		s.nonceCache.Resize(s.config().NonceCacheSize) // Would probably be better in a config hook but this is basically free
```

Comment:
consider moving into createBlock, just before BeginNewBlock. Not really different, but it's a little easier to see that it's not in parallel with using the nonceCache.